### ISSUE TITLE
the `block` command is broken and why does it exist?

### ISSUE THREAD

**Issue by krader1961** [fish-shell/fish-shell#3517]
While working on eliminating `missing break in switch statement` warnings from oclint I noticed that `builtin_block()` in *src/builtin.cpp* is badly broken. The `switch` block in the else branch of `if (erase) {` falls through to the `default:` block which kills the process. All you have to do to see the problem is type `block -g wtf` and observe your shell die. I recommend you first type `fish` so you don't lose your login shell.

AFAICT this command isn't used anywhere in the fish project.

@ridiculousfish, Care to comment? I can sort of see the point of having such a command but the current implementation is broken and there do not appear to be any unit tests for its correct behavior.

**Comment by krader1961** [comment]
This issue is a poster child example for why we should be paying attention to warnings from tools like cppcheck and oclint.

**Comment by krader1961** [comment]
Okay, I see that while the code in question is bad form it's not "badly broken". The problem is that two of the three `case` blocks in the `switch` statement do not have a `break;` statement. The first of them falls through to the second. Which is harmless but apparently by accident and is very bad form. The second is harmless because it was the last `case` block before I added a `default` block. Again, bad form (and oclint was also warning about that). I simply failed to notice the lack of `break;` statements. Once I added those the problem went away.

So the lesson is: pay attention to all the lint warnings.
